### PR TITLE
For #20762 - Fixes NPE crash

### PR DESCRIPTION
**Pull request by codrut-topliceanu**
Keeps the `_binding` from being set to null `onDestroy`. While not ideal it will prevent NPEs from happening on coroutines that live longer than the view.

For https://github.com/mozilla-mobile/fenix/issues/21418
### Pull Request checklist

- [x] **Tests**: This PR includes not tests as it's only a minor change
- [x] **Screenshots**: This PR includes no screenshots or GIFs as the changes aren't visible
- [x] **Accessibility**: The code in this PR does not include any user facing features.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/addons/AddonsManagementFragment.kt [hunk 1]
```diff
@@ -90,7 +90,9 @@ class AddonsManagementFragment : Fragment(R.layout.fragment_add_ons_management)
         super.onDestroyView()
         // letting go of the resources to avoid memory leak.
         adapter = null
-        _binding = null
+        // Intentionally leaking the binding since it is used when the async operation returns
+        // It will ultimately be destroyed when the Fragment instance is destroyed.
+        // _binding = null
```
Comment (pocmo) [inline app/src/main/java/org/mozilla/fenix/addons/AddonsManagementFragment.kt:95]: This is not great. Can't we fix the async operation(s) to use a nullable type?

**Review by csadilek — COMMENTED** [review]
Thanks @codrut-topliceanu for taking this on so quickly! We should also uplift the fix so we don't ship the crash to release, but let's land in Nightly and verify first.

I also think we should check for null in those callbacks or using `binding?.` instead. To prevent a bigger refactor, I think it's also OK to use `_binding` in those callbacks which is already `nullable` with a comment to explain why :).

**Comment by Mugurell** [comment]
Drive by comment:
It seems that the crash stems from 
https://github.com/mozilla-mobile/fenix/blob/7070bb5607f5ac9fd0825ab21e7abba675f98bdc/app/src/main/java/org/mozilla/fenix/addons/AddonsManagementFragment.kt#L84-L85
which tries to update a fragment from the backstack, which in that already gone through `onDestroyView`.
If we are to continue to null the binding in `onDestroyView` and check it for null here then we'd break this functionality.
I'd say we either "leak" the binding past `onDestroyView` either refactor the functionality .. which it isn't clear how it works.

**Comment by csadilek** [comment]
> If we are to continue to null the binding in onDestroyView and check it for null at line 85 then we'd break the current functionality which needs that to not be null.

Hm I don't understand what functionality would break here though. If the view is destroyed we'd simply do nothing if we check for null or use `binding?.` which means we wouldn't hide the `addonProgressOverlay` but since the view was destroyed it wouldn't be visible anymore anyway?

So, it seemed fine to me, but either way we should land a fix so we can uplift next week and get rid of the crash :).

**Review by csadilek — APPROVED** [review]
Thanks, let's 🚢  and address the crashes. Curious if you have suggestions how to improve this fragment in general. If you have them, let's file an issue, there's probably a cleaner way we could handle those async ops.

**Pull request merged**